sumomongodbatlascollector/main.py

```diff
@@ -209,18 +228,45 @@ def run(self):
                         self.log.error(f"API Type: {api_type} thread generated an exception: {exc}", exc_info=True)
                     else:
                         self.log.info(f"API Type: {api_type} thread completed {obj}")
+                end_time = time.time()
+                _, peak_memory = tracemalloc.get_traced_memory()
+                tracemalloc.stop()
+
+                self.log.info(f"Total execution time: {end_time - start_time:.2f} seconds")
+                self.log.info(f"Peak memory usage: {peak_memory / 1024 / 1024:.2f} MB")
             finally:
                 self.stop_running()
                 self.mongosess.close()
         else:
             if not self.is_process_running(["sumomongodbatlascollector"]):
                 self.kvstore.release_lock_on_expired_key(self.SINGLE_PROCESS_LOCK_KEY, expiry_min=10)
 
+    def execute_api_with_logging(self, apiobj):
+        api_type = str(apiobj.__class__.__name__)
+        start_time = time.time()
+        start_memory = tracemalloc.get_traced_memory()[0]
+        try:
+            result = apiobj.fetch()
+        except Exception as exc:
```

Comment:
not needed we can remove this exeception

Reply:
Removed this exception,now we are doing time and process memory logging with context manager